script/metalsmith/nonceTransformer.js

```diff
@@ -0,0 +1,42 @@
+const jsdom = require('jsdom');
+const path = require('path');
+
+function generateNewId(existingIds) {
+  const newId = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 10);
+  if (!existingIds.has(newId)) {
+    existingIds.add(newId);
+    return newId;
+  }
+  return generateNewId(existingIds);
+}
```

Comment:
Not sure if it's exactly what you're looking for but Lodash has the [`uniqueId`](https://lodash.com/docs/4.17.5#uniqueId) function.